### PR TITLE
chore: Update dependencies and just recipes

### PR DESCRIPTION
**Pull request by pedroheyerdahl**
This PR updates project dependencies to resolve conflicts, remove unused libraries, and introduce just recipes for SQLMesh transformations and the complete pipeline (a combination of ingest, transform, and upload steps).

### Dependencies
- Set version constraints for core packages:
  - numpy < 2.0.0
  - pandas < 3.0.0
  - sqlmesh == 0.122.3
- Removed unnecessary dependencies

### Just Recipes
Added new commands:
````bash
just transform # Run SQLMesh transformations
just etl # Run complete pipeline

````
### Other Changes
- Updated .gitignore to exclude system files.
- Enhanced setup documentation for better clarity and onboarding.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -145,8 +189,8 @@ This project is licensed under the MIT License - see the [LICENSE](LICENSE) file
 ## Contact
 
 Mirian Lima - mirian.lima@un.org
```
Comment (db-tycoon-stephen) [inline README.md:191]:
```suggestion
Mirian Lima (Project Sponsor) - mirian.lima@un.org
Stephen Sciortino (Principal Engineer, Consultant) - stephen.sciortino@un.org; stephen@databasetycoon.com
```

**Review by db-tycoon-stephen — CHANGES_REQUESTED** [review]
Hey @pedroheyerdahl I got an issue installing the requirements. I'm on 3.12.3. We may need to set a version max in the readme.
Which did you use to run this?
image

**Comment by db-tycoon-stephen** [comment]
I resolved it by running the install with python 3.11. Could you add <3.12 to the section in the readme?
Otherwise this is ready to merge!

**Comment by pedroheyerdahl** [comment]
> I resolved it by running the install with python 3.11. Could you add <3.12 to the section in the readme? Otherwise this is ready to merge!

@db-tycoon-stephen nice catch! version constraint is now clarified in the readme. ty!

**Review by db-tycoon-stephen — APPROVED** [review]
Looks good!

**Pull request merged**